### PR TITLE
New Servicebus RP version 2017-04-01

### PR DESCRIPTION
**Pull request by v-Ajnava**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by vishrutshah — CHANGES_REQUESTED** [review]
@v-Ajnava Could you please follow the guidelines outlined https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md#filenames-and-folder-structure .

Please make sure you create 3 commits. One with just copy of the old swagger into the new folder and second with the changes to help us review the changes only and third one with just example files. That's greatly help us in reviewing the changes. Thanks.

**Comment by vishrutshah** [comment]
@v-Ajnava Do you know why you are being tagged as cla not signed?

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 1]
```diff
@@ -1650,113 +1872,295 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please add `default` response

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 2]
```diff
@@ -1650,113 +1872,295 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content"
+          },
+          "200": {
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please add `default` response

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 3]
```diff
@@ -1650,113 +1872,295 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: `default` response please

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 4]
```diff
@@ -1650,113 +1872,295 @@
...
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "TrackedResource": {
+      "properties": {
+        "location": {
+          "type": "string",
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please add `x-ms-mutability` Reference: https://github.com/Azure/azure-rest-api-specs-pr/blob/master/common-types/resource-management/v1/types.json#L23

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 5]
```diff
@@ -2565,31 +2835,218 @@
...
+          "$ref": "#/definitions/Action",
+          "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+        },
+        "filterType": {
+          "$ref": "#/definitions/FilterType",
+          "description": "Filter type that is evaluated against a BrokeredMessage."
+        },
+        "sqlFilter": {
+          "$ref": "#/definitions/SqlFilter",
+          "description": ""
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please provide complete `description`

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 6]
```diff
@@ -2565,31 +2835,218 @@
...
+        },
+        "correlationFilter": {
+          "$ref": "#/definitions/CorrelationFilter",
+          "description": ""
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please provide complete `description`

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 7]
```diff
@@ -2565,31 +2835,218 @@
...
+		"sqlExpression": {
+          "type": "string",
+          "description": "SQL expression. e.g. MyProperty='ABC'"
+        },
+        "compatibilityLevel": {
+          "format": "int32",
+          "type": "integer",
+          "description": "This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20."
+        },
+		"requiresPreprocessing": {
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please format the json file

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 8]
```diff
@@ -2565,31 +2835,218 @@
...
+          "type": "boolean",
+		  "default": true,
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please format the json file

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 9]
```diff
@@ -2094,20 +2094,44 @@
       "properties": {
         "id": {
           "type": "string",
+		  "readOnly": true,
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: please format the json file

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 10]
```diff
@@ -0,0 +1,3152 @@
...
+      },
+      "delete": {
+        "tags": [
+          "Rules"
+        ],
+        "operationId": "Rules_Delete",
+        "x-ms-examples": {
+          "RulesDelete": { "$ref": "../examples/Rules/RuleDelete.json" }
+        },
+        "description": "Deletes an existing rule.",
```
Comment (vishrutshah) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Please add description as they are vital part of the documentation for each service. 

There are `11` warning of  `DescriptionAndTitleMissing `. Please fix.

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/243507256#L1291

**Review by vishrutshah — CHANGES_REQUESTED** [review]

**Comment by vishrutshah** [comment]
@v-Ajnava Please also verify `TrackedResourceListByImmediateParent` RPCViolation category and let us know. Thanks!

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 9 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Resolved.

File: arm-servicebus/2017-04-01/swagger/servicebus.json [hunk 11]
```diff
@@ -1650,113 +1872,295 @@
...
+        ],
+        "operationId": "Rules_CreateOrUpdate",
+        "x-ms-examples": {
+          "RulesCreateOrUpdate": { "$ref": "../examples/Rules/RuleUpdate.json" }
         },
-        "tags": {
-          "type": "object",
-          "additionalProperties": {
-            "type": "string"
+        "description": "Creates a new rule",
```
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Updated description, resolved.

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 1 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Added default response, resolved

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 2 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Added default response, resolved

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 3 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Added default response, resolved

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 4 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: added x-ms-mutability and resolved.

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 5 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: added description

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 6 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: added description

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 10 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: Added descriptions

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 7 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: formatted json.

File: arm-servicebus/2017-04-01/swagger/servicebus.json (hunk 8 above)
Comment (v-Ajnava) [inline arm-servicebus/2017-04-01/swagger/servicebus.json]: formatted json.

**Comment by v-Ajnava** [comment]
@vishrutshah, we have completed the CLA document, can you please verify it again.

**Comment by vishrutshah** [comment]
@v-Ajnava Closing and re-opening PR to make sure CLA recognizes you :)

**Comment by msftgits** [comment]
Closing and reopening to see if it updates the CLA status appropriately.
-JW

**Review by vishrutshah — CHANGES_REQUESTED** [review]
@v-Ajnava Thanks for addressing the comments. I now see that you only have 2 cateogries of warnings:

1. `TrackedResourceListByImmediateParent`
2. `EnumInsteadOfBoolean`

I assume that the service team has made a decision to keep those boolean instead of Enums. That is totally fine. What about the few warnings of type `TrackedResourceListByImmediateParent`. Did get a chance to look into them? https://travis-ci.org/Azure/azure-rest-api-specs/jobs/245137564#L1190

Let me know!

**Comment by v-Ajnava** [comment]
@vishrutshah, the 'TrackedResourceListByImmediateParent' warnings for 'queues' and 'topics' is for SBNamespaces. I think its because of the prefix of 'SB' added to namespace (SBNamespace) as its reserved word and in SDK the generated code it was renamed as NamespaceModel. similar way for 'subscription' as 'topics' also has a prefix of 'SB'. 

for Authorization rules: 
The Authorization rule is child resource of Namespace, Queue and Topic. 
Namespaces -> Authorizationrule 
Queue-> Authorizationrule
Topic-> Authorizationrule

We have name the Authorization rule operations for namespace as ‘Namespaces_ListAuthorizationRules’ which list Authorizationrules under namespaces, similarly ‘Namespaces_ListKeys’ which list the Keys and connection strings of the specified Authorizationrules. 

In the SDK generated code we maintain the hierarchy as 

Client.Namespaces.ListAuthorizatiorules(…) 
Client.Namespaces.ListKeys(…)

If we  change the OperationId to suggested one, as ‘AuthorizationRules_ListByNamespace’ the SDK will be as below and bit confusing. 

Client.Authorizationrules.ListByNamespace(…)
client.AccessKeys.ListByNamespaces(…)

In the Previous RP release we had offline discussion with reviewers about the OperationId naming convention for the Authorizationrules and continued with same naming conventions. 

forwarding a email thread regarding the same. 

please let me know if you need more details or we can have a discussion.

**Review by vishrutshah — APPROVED** [review]
Thanks @v-Ajnava for the detailed explanation. Please forward me that email thread for my reference. Thanks!

**Comment by vishrutshah** [comment]
@v-Ajnava Thanks for the email thread.

**Pull request merged**